numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2058,9 +2054,8 @@ class ManagedOwnedPointer(OwnedPointer, mviewbuf.MemAlloc):
     pass
 
 
-class Stream(object):
-    def __init__(self, context, handle, finalizer, external=False):
-        self.context = context
+class Stream:
+    def __init__(self, handle, finalizer=None, external=False):
```

Comment:
There is unfortunately public code using the `Stream` constructor here with the context argument. It looks like most of it is stale and pinned to old versions of Numba, but we should maybe be a bit mindful.

@gmarkall what are your thoughts?

Reply:
We should probably lay out somewhere what constructors are public, since hand constructing these doesn't appear to be part of any explicit public API, just the one you get from the default Python conventions.

Reply:
Can you highlight the code that's using the constructor explicitly? The way it's presented in the documentation is maybe not ideal, but users are directed to use one of:

* `cuda.stream()`
* `cuda.default_stream()`
* `cuda.legacy_default_stream()`
* `cuda.per_thread_default_stream()`
* `cuda.external_stream(ptr)`

to construct a `Stream` object rather than using the constructor directly: https://nvidia.github.io/numba-cuda/reference/host.html#stream-management